### ISSUE TITLE
Add support for kqueue in multiplexer

### ISSUE THREAD

**Issue by mavam** [actor-framework/actor-framework#533]
CAF currently doesn't leverage the kqueue scalable event notification system native to several BSD flavors. Support for kqueue would improve the multiplexing performance on platforms offering it.

**Comment by josephnoir** [comment]
Yes! Wanted to take a look at that, too. (Although, I'm missing the time to take a look at that in the next weeks. )

**Comment by mavam** [comment]
I've added kqueue support in branch `topic/kqueue`. Still needs more testing.

**Comment by mavam** [comment]
All tests passing on FreeBSD and MacOS. Green light from my side to merge into develop.

**Comment by Neverlord** [comment]
@mavam did you merge this into `develop` yet? I don't remember a PR for this feature.

**Comment by mavam** [comment]
Sorry, this got lost somehow. Let's discuss implementation details in the new PR #575.

**Comment by Neverlord** [comment]
@mavam Is it still in progress?

**Comment by mavam** [comment]
The implementation of #575 is practically complete, but I currently lack the cycles lately to hunt down a bug that shows up only once during 100 invocations.

You've closed #575 a few weeks ago so I'm closing this one as well. Let's reopen #575 when the time is ripe.
